File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.c
#include <stdio.h>
#include <string.h>

char * reversePrefix(char * word, char ch) {
    int len = strlen(word);
    int index = -1; // 문자 위치 인덱스

    // 문자 위치 찾기
    for (int i = 0; i < len; ++i) {
        if (word[i] == ch) {
            index = i;
            break;
        }
    }

    // 문자를 찾지 못한 경우 원래 문자열 반환
    if (index == -1) {
        return word;
    }

    // 문자열 뒤집기
    for (int i = 0; i <= index / 2; ++i) {
        char temp = word[i];
        word[i] = word[index - i];
        word[index - i] = temp;
    }

    return word;
}

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        for i in range(len(word)):
            if word[i]==ch:
                if i==0:
                    return word
                elif i == 1:
                    return word[1]+word[0]+word[2:]
                else:
                    new=word[:i]
                    return word[i]+new[::-1]+word[i+1:]
        return word